Fixing Everything at Once

monsieuremre opened the pull request:
* Fix image conversion loop header

* Add readme section to mention known limitations on other operating systems than windows

* Add script section to convert inline images to image blocks (needs confirmation)

* Add script section to explicitly remove bibliography from section numeration if that is not already the case

monsieuremre commented:
Fixed literally everything, including but not limited to:

* Figures, images, tables, chapters and anything in between are now referenced throughout the document accurately and consistently.
* Various breakages with chapters and sections that normally required manual fixing are now handled in the script, no manual fixes required for them at all anymore.
* Tables with appendices that are special and merged are now formatted and fixed completely within the script and they do not require any manual fixing at all anymore.
* Randomly occuring characters that disrupt the expect flow of the text are now removed within the script automatically.
* There are some changes in the old functions and one of them is deleted completely. There was no other way if we wanted to set the references appropriately. The removed sections' functionality is implemented in other, newly added functions, if it was necessary.

Now, with discretion, the script should handle *anything and everything fixing related*. That is, the manual fixing section should not be required anymore. No not merge yet, needs testing.

I tested all the individual functions and code blocks manually and they all do work perfectly. Whether or not applying all of them consecutively would cause some kind of unexpected behavior or breakage is not known, because I did not conduct any tests on the module as a whole. There might also be some side-effects that I missed that might need fixing, but for the time, there seems to be none.

I have readjusted the sequence in which we call the fixing functions to accomodate for the new functionality, and to ensure that new functions would not break. This is important because some of them have to be called before others to perform the fixes. e. g. setting the reference links needs to be done after getting rid of the <<>> stuff. There are more like this that I already handled. There might be more of these that I could not think of, these will be discovered and fixed once I test the standalone module as a whole.

monsieuremre commented:
Needless to say, I only addressed the manual fixes that you mentioned to me. Now I noticed there are two little ones that I missed:
* Setting document attributes
* Removing the various tables in the beginning.

This should not be hard to add too.

monsieuremre commented on word2asciidoc/formatting.py:
```diff
@@ -110,5 +85,127 @@ def add_links_to_bibliography(content, keys):
+        r'\[.*\](\n)+\|\===\n((\|.*\n)+?)\|Inherits from:.*\|\n',
+        repl,
+        content)
+    return content
```
New push makes sure it works on all documents

monsieuremre commented on word2asciidoc/formatting.py:
```diff
@@ -110,5 +85,127 @@ def add_links_to_bibliography(content, keys):
+        r'\n(=)*( )*(image:.*\])(.*)(\n)+(\[#_(Ref|Toc).*\])(.*(\n))(.*(\n))',
+        repl_1,
+        content)
+    return content
```
all addressed

monsieuremre commented:
Tested everything at once. There were problems, maybe unsurprisingly. They are all fixed. I also simplified the code wherever possible and split it into many modular pieces, and added comments for readibility, as well as renamed function as suggested.

monsieuremre commented:
Please see [this document](https://github.com/monsieuremre/aas-specs/blob/asciidoc_master/docs/AASiD/AASiD_3_iec61360/IDTA-01003-a-3-0_SpecificationAssetAdministrationShell_Part3a_DataSpecification_IEC61360.adoc). It is generated completely and solely using the script. No manual fixes. Please take a look and report what is missing, what needs manual fixing, what is broken (if I missed anything). I will try to further develop the script to handle the fixes that you think are missing.

zrgt commented:
Multiple things are broken:
- Images are not defined as blocks
- Images are used as titles
- Image/Table/Subclause References do not work
- Formating bugs
- Others

zrgt commented:
- Please install PyCharm and AsciiDoc Viewer Plugin. 
- If you have generated an asciidoc file, try to generate an html file (one click in asciidoc plugin for pycharm) from that.
- Compare the content of the generated HTML with the source DOCX and check if everything is alright. 
- Do think twice before writing a script for some issue, do the script only if you are sure, that the issue can come in various documents and your script is general enought to solve that issue in different documents

monsieuremre commented:
Ok. The table splitting works on most documents, not always, and never causes breakage. So leaving it as is. Sometimes manual table splitting is still needed. Reference works in any case. Ordering might sometimes need manual fixing. The amount of manual fixing is minimized to its limits so that not even half the manual fixing isrequired as before, but unfortunately eliminating it completely is not possible because there is too big of a deviation between documents and their structure.

Whatever there is, it is tested and documented and simplified. Readme is also updated. Merging.
